launcher/modplatform/flame/FlameCheckUpdate.cpp

```diff
@@ -121,7 +123,9 @@ void FlameCheckUpdate::getLatestVersionCallback(Resource* resource, std::shared_
                 old_version = tr("Unknown");
         }
 
-        auto download_task = makeShared<ResourceDownloadTask>(pack, latest_ver.value(), m_resourceModel);
+        QDir target_dir(resource->fileinfo().absolutePath());
+        auto download_task = makeShared<ResourceDownloadTask>(pack, latest_ver.value(), m_resourceModel, true, target_dir.absolutePath(),
```

Comment:
Same as modrinth update this will download to the main directory and keep the old files if they need update and are in subfolder